File: commands/jamie.js
const { EmbedBuilder, AttachmentBuilder, SlashCommandBuilder} = require("discord.js");
module.exports = {
    data: new SlashCommandBuilder()
        .setName('jamie')
        .setDescription('Gives Jamie info')
        .addSubcommand(profile => profile
            .setName('profile')
            .setDescription('Shows Jamie info or something'))
        .addSubcommand(photo => photo
            .setName('photo')
            .setDescription('sends the Jamie photo')),
    async execute(interaction, client){
        await interaction.deferReply();

        const closeFriends = [];

        let role = client.guilds.cache.get('976149800447770624').roles.cache.get('976175903371571220')
        let interactionGuildPromise = await interaction.guild.members.fetch()

        if(typeof role === "object"){
            role.members.map(async member => {
                if(typeof await interactionGuildPromise.find(f => f.id === member.id) === 'object') closeFriends.push(member.user.tag)
            })
        }

        setTimeout(async () => {
            const jamieObject = await client.users.fetch("394127601398054912");
            const isJamieObject = typeof jamieObject === "object";
            if(interaction.options.getSubcommand() === 'profile'){
                let jamie = {
                    heIS: (isJamieObject),
                    tag: (isJamieObject) ? jamieObject.tag : "Jamie#8409",
                    id: (isJamieObject) ? jamieObject.id : "394127601398054912",
                    pfp: (isJamieObject) ? jamieObject.avatarURL() : "https://cdn.discordapp.com/avatars/394127601398054912/c7a08756f08ff4fa9f51cf5f63f017d0.png?size=4096"
                }

                client.db.all(`SELECT count(*) FROM Worshippers`, async (err, WorshippersCountRow) => {
                    client.db.all(`SELECT * FROM JamieInfo WHERE Key = 'messageNum'`, async (err, MessageNumRow) => {
                        client.db.all(`SELECT * FROM JamieInfo WHERE Key = 'LastMessage'`, async (err, LastMessageRow) => {
                            const embed = new EmbedBuilder()
                                .setTitle('Jamie')
                                .setDescription(`**Jamie\'s tag:** ${jamie.tag}\n**Jamie\'s id:** ${jamie.id}\n\nHe lives in UK even tho he says he doesn\'t`)
                                .addFields(
                                    {"name": 'Jamie\'s close friends', "value": (closeFriends.length !== 0) ? closeFriends.toString().replaceAll(',', '\n') : "No close friends on this server."},
                                    {"name": 'Jamie\'s bot stats', "value": `**Worships:** ${WorshippersCountRow[0]['count(*)']}\n**Counted messages:** ${MessageNumRow[0].Value}\n**Last message:** ${LastMessageRow[0].Value}`}
                                )
                                .setThumbnail(jamie.pfp)
                                .setColor(Math.floor(Math.random() * 0xFFFFFF))

                            if(jamie.heIS === false) embed.setFooter({text: 'The info may not be up to date since the user can\'t be fetched.'})

                            interaction.editReply({embeds: [embed]})
                        })
                    })
                })
            }else if(interaction.options.getSubcommand() === 'photo'){
                try{
                    const photo = new AttachmentBuilder('./images/helloJamie.gif', 'helloJamie.gif')
                    await interaction.channel.send({files: [photo]})
                    interaction.deleteReply();
                }catch{
                    interaction.editReply({content: `Can't send images in this channel`})
                }
            }
        }, 500)
    }
}

File: Commands/Jamie.cs
﻿using Discord;
using Discord.Interactions;
using JamieWorshipper.Handlers;
using JetBrains.Annotations;

namespace JamieWorshipper.Commands;

[Group("jamie", "Jamie related commands group.")]
public class Jamie : InteractionModuleBase<SocketInteractionContext>
{
    [SlashCommand("profile", "View jamie profile."), CommandCooldown(15), UsedImplicitly]
    public async Task JamieProfileAsync()
    {
        await DeferAsync();
        
        IRole closeFriendsRole = Config.MainGuild.Roles.First(r => r.Id == Config.CloseFriendsRole.Id);
        List<List<object>> dbData = DataBase.RunSqliteCommandAllRows(@"
        SELECT count(*) FROM Worships UNION ALL
        SELECT InfoValue FROM JamieInfo WHERE InfoKey = 'MessageNum' UNION ALL
        SELECT InfoValue FROM JamieInfo WHERE InfoKey = 'LastMessage';        
        ");

        string closeFriends = (await Config.MainGuild.GetUsersAsync())
            .Where(u => u.RoleIds.Any(s => s == closeFriendsRole.Id))
            .Aggregate("", (current, user) => current + $"{user}, ")[..^2];

        EmbedBuilder embed = new EmbedBuilder()
            .WithTitle("Jamie")
            .WithDescription($"**Jamie's tag:** {Config.Jamie}\n**Jamie's id:** `{Config.Jamie.Id}`\n\nHe lives in UK even tho he says he doesn't")
            .WithFields(
                new EmbedFieldBuilder().WithName("🔹 Jamie's close friends").WithValue(closeFriends),
                new EmbedFieldBuilder().WithName("🔹 Jamie's bot stats").WithValue($"**Worships:** {dbData[0][0]}\n**Counted messages:** {dbData[1][0]}\n**Last message:** {dbData[2][0]}")
            )
            .WithThumbnailUrl(Config.Jamie.GetAvatarUrl())
            .WithColor(RandomColor());

        await ModifyOriginalResponseAsync(r => r.Embed = embed.Build());
    }

    [SlashCommand("gif", "Jamie gifgifgifgif."), CommandCooldown(30), RequireBotPermission(ChannelPermission.AttachFiles), UsedImplicitly]
    public async Task JamiePhotoAsync([Summary("Send", "Indicate whether send the gif to jamie or not.")]bool send = false)
    {
        await DeferAsync();

        if (!File.Exists("./build/JamieGif.gif"))
        {
            async Task CreateFileTask()
            {
                await using FileStream fs = File.Create("./build/JamieGif.gif");
                using HttpClient client = new HttpClient();
                await fs.WriteAsync(await client.GetByteArrayAsync("https://cdn.memw.es/helloJamie.gif"));
            }

            await CreateFileTask();
        }

        await Context.Channel.SendFileAsync(filePath: "./build/JamieGif.gif");
        await ModifyOriginalResponseAsync(r => r.Content = "Loaded.");
        if (send)
            try
            {
                if (Context.User.IsOnBlackList())
                {
                    await FollowupAsync("🙅 You are in the blacklist, you cannot send anything to jamie. 🚫", ephemeral: true);
                    return;
                }

                await Config.Jamie.SendFileAsync("./build/JamieGif.gif");
                await FollowupAsync("The gif was sent to jamie", ephemeral: true);
            }
            catch
            {
                await FollowupAsync("An error occurred, the gif was not sent, maybe jamie blocked the bot, but you can still enjoy your gif without sending it.", ephemeral: true);
            }

        await DeleteOriginalResponseAsync();
    }
}